primitives/src/vid/advz.rs

```diff
@@ -279,24 +306,35 @@ where
     V::MembershipProof: Sync + Debug, /* TODO https://github.com/EspressoSystems/jellyfish/issues/253 */
     V::Index: From<u64>,
 {
-    /// Same as [`VidScheme::disperse`] except `payload` is a slice of
+    /// Same as [`VidScheme::disperse`] except `payload` iterates over
     /// field elements.
     pub fn disperse_from_elems<I>(&self, payload: I) -> VidResult<VidDisperse<Self>>
     where
         I: IntoIterator,
         I::Item: Borrow<P::Evaluation>,
     {
-        let domain = P::multi_open_rou_eval_domain(self.payload_chunk_size, self.num_storage_nodes)
-            .map_err(vid)?;
-
         // partition payload into polynomial coefficients
         // and count `elems_len` for later
         let elems_iter = payload.into_iter().map(|elem| *elem.borrow());
         let mut elems_len = 0;
         let mut polys = Vec::new();
         for coeffs_iter in elems_iter.chunks(self.payload_chunk_size).into_iter() {
```

Comment:
These sets are not equal. The first is a subset of the second. (We use a larger FFT domain in line 341 because we are evaluating a degree `d` polynomial at `>d` points.)

It's true that `d` items in `elems_iter` are the same as poly evaluations in line 341. So yes, there's some waste here. But:
1. We're using FK23 to get poly evaluations for all the shares. I don't know how to avoid re-computing some shares unless with ditch FK32 and replace it with something asymptotically slower.
2. This PR is intended as a temporary stop-gap to unlock progress on namespaces. The eventual goal is to implement KZG in eval form, at which time we will no longer be doing any wasteful re-computation.